modules/homa-lite/src/lib.rs

```diff
@@ -267,6 +281,23 @@ pub mod module {
 			}
 			current_weight
 		}
+
+		fn on_initialize(n: T::BlockNumber) -> Weight {
+			// Update the total amount of Staking balance by acrueing the interest periodically.
+			let interest_rate = Self::staking_interest_rate_per_update();
+			if !interest_rate.is_zero()
+				&& n.checked_rem(&T::StakingUpdateFrequency::get())
+					.unwrap_or_else(One::one)
```

Comment:
this is a bit confusing

could be 

`n.checked_rem(&T::StakingUpdateFrequency::get()).map_or(false, Zero::is_zero)`